consensus/istanbul/backend/backend.go

```diff
@@ -992,3 +873,34 @@ func (sb *Backend) retrieveUncachedValidatorConnSet() (map[common.Address]bool,
 	logger.Trace("Returning validator conn set", "validatorsSet", validatorsSet)
 	return validatorsSet, nil
 }
+
+func (sb *Backend) sendForwardMsgToProxy(finalDestAddresses []common.Address, ethMsgCode uint64, payload []byte) error {
+	logger := sb.logger.New("func", "sendForwardMsgToProxy")
+	if sb.proxyNode.peer == nil {
```

Comment:
This is definitely outside the scope of this PR, but I've always wondered if we risk the peer being set and passing a nil check, and then the peer disconnects and becomes nil, and then `sb.proxyNode.peer.Send` panics. Do you have any thoughts on that?

Reply:
Ya, I'm planning to address this in the multi-proxy PR (which includes a major refactoring of the proxy code).

The plan is for the proxy handler to be the single thread set and unset the `peer` field, as well as send messages to that `peer`.  So that will handle any race conditions.